pkg/epp/requestcontrol/director.go

```diff
@@ -138,17 +140,18 @@ func (d *Director) HandleRequest(ctx context.Context, reqCtx *handlers.RequestCo
 	ctx = log.IntoContext(ctx, logger)
 	logger.V(logutil.DEBUG).Info("LLM request assembled")
 
-	// --- 2. Admission Control check --
-	if err := d.admitRequest(ctx, requestCriticality, reqCtx.FairnessID); err != nil {
-		return reqCtx, err
-	}
-
-	// --- 3. Call Scheduler (with the relevant candidate pods) ---
+	// Get candidate pods for scheduling
 	candidatePods := d.getCandidatePodsForScheduling(ctx, reqCtx.Request.Metadata)
 	if len(candidatePods) == 0 {
 		return reqCtx, errutil.Error{Code: errutil.ServiceUnavailable, Msg: "failed to find candidate pods for serving the request"}
 	}
-	result, err := d.scheduler.Schedule(ctx, reqCtx.SchedulingRequest, candidatePods)
+
+	// Admission Control check
+	if err := d.admitRequest(ctx, candidatePods, requestCriticality, reqCtx.FairnessID); err != nil {
```

Comment:
Maybe we could move `admiRequest` logic into `getCandidatePodsForScheduling`. Intuitively, the `admiRequest` function should be called before the `getCandidatePodsForScheduling` function.

Reply:
the main logic currently in `admitReqeust` is to check saturation of the system.
it's not relevant to check saturation of all the pods but only of the candidate pods (If we use subset filtering to specify list of pods and all of them are saturated, it's not helpful that other pods are available).

therefore, `getCandidatePods` is done before the `admitRequest` function.

think about it this way - 
`getCandidatePods` func returns the set of pods that are relevant for current request.
`admitRequest` func check if any of the pods from the above set can actually serve the request, or if all of them are saturated.